### PR TITLE
implemented numpy frontend asarray function and its tests

### LINKED ISSUE #4701: asarray

### PR DESCRIPTION
**Pull request by ronnith24**
Close #4701

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    as_variable,
+    num_positional_args,
+    fw,
+):
+    dtype, a = dtype_and_a
+    helpers.test_frontend_function(
+        input_dtypes=dtype,
+        as_variable_flags=as_variable,
+        num_positional_args=num_positional_args,
+        native_array_flags=False,
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py:31]: please take a look at the example given in the deep dive document: https://lets-unify.ai/ivy/deep_dive/17_ivy_frontends_tests.html

**Comment by Daniel4078** [comment]
Also, please try to fix these lint formatting errors:
ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py:8:1: E302 expected 2 blank lines, found 1
ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py:18:89: E501 line too long (102 > 88 characters)

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py (hunk 1 above)
Comment (ronnith24) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_existing_data.py]: I think have made the nescessary changes, please let me know incase of further changes.

**Comment by ronnith24** [comment]
I have also fixed the linting errors, plz let me know incase of further changes

**Review by Daniel4078 — APPROVED** [review]
It seems like there is no error in your code now and all the relevant tests pass. Good work.

**Pull request merged**